pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -133,12 +133,13 @@ def on_load_checkpoint(self, callback_state: Dict[str, Any]):
         self.patience = callback_state['patience']
 
     def on_validation_end(self, trainer, pl_module):
-        if trainer.running_sanity_check:
+        from pytorch_lightning.trainer.states import TrainerState
```

Comment:
To avoid a circular import

Reply:
It shall have only enums, so there shall be no cycle... So the wrong importing is elsewhere

Reply:
IDK how it could be solved without moving files. this is the dependency in case you have an idea:

callbacks -> early_stopping -> trainer.state -> trainer -> callback_connector -> callbacks

Reply:
I see 3 options:

Reply:
Not sure I like the option, it will make the code complicated to enable this simple import there.
Might be better to live it like that, but it should be exceptional.